mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb

```diff
@@ -11547,3 +11547,128 @@ resource "google_container_cluster" "primary" {
 }
 `, name, networkName, subnetworkName)
 }
+
+func TestAccContainerCluster_storagePoolsWithNodePool(t *testing.T) {
+	t.Parallel()
+
+	cluster := fmt.Sprintf("tf-test-cluster-%s", acctest.RandString(t, 10))
+	np := fmt.Sprintf("tf-test-nodepool-%s", acctest.RandString(t, 10))
+	networkName := acctest.BootstrapSharedTestNetwork(t, "gke-cluster")
+	subnetworkName := acctest.BootstrapSubnet(t, "gke-cluster", networkName)
+	location := envvar.GetTestZoneFromEnv()
+
+	storagePoolNameURL := acctest.BootstrapComputeStoragePool(t, "basic-1", "hyperdisk-balanced")
+	storagePoolResourceName, err := extractSPName(storagePoolNameURL)
+	if err != nil {
+		t.Fatal("Failed to extract Storage Pool resource name from URL.")
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckContainerNodePoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_storagePoolsWithNodePool(cluster, np, networkName, subnetworkName, storagePoolResourceName, location),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("google_container_cluster.storage_pools_with_node_pool", "node_pool.0.node_config.0.storage_pools.0", storagePoolResourceName),
+				),
+			},
+			{
+				ResourceName:            "google_container_cluster.storage_pools_with_node_pool",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"deletion_protection"},
+			},
+		},
+	})
+}
+
+func testAccContainerCluster_storagePoolsWithNodePool(cluster, np, networkName, subnetworkName, storagePoolResourceName, location string) string {
+	return fmt.Sprintf(`
+resource "google_container_cluster" "storage_pools_with_node_pool" {
+  name               = "%[1]s"
+  location           = "%[6]s"
+  deletion_protection = false
+  network    = "%[3]s"
+  subnetwork    = "%[4]s"
+  node_pool {
+    name = "%[2]s"
+    initial_node_count = 1
+    node_config {
+      machine_type = "c3-standard-4"
+      image_type = "COS_CONTAINERD"
+      storage_pools = ["%[5]s"]
```

Comment:
Got it. Thanks for confirming! I'll update the bootstrap function later. I initially intended for it to return the long form name but noticed it currently returns the URL instead.

Reply:
I think this is now causing problems since it seems to have moved from v1 to beta? Maybe there's a reason it wasn't done this way to start with, but threw up https://github.com/GoogleCloudPlatform/magic-modules/pull/11832 for consideration